compiler/accelerators/rocc.py

```diff
@@ -35,52 +37,84 @@ def lower_acc_setup(
         setup_op: acc.SetupOp, acc_op: acc.AcceleratorOp
     ) -> Sequence[Operation]:
         xcustom_acc = 3  # hardcoded to 3 for now
-        vals = create_pairs(setup_op, acc_op.field_names())
+        vals = create_pairs(setup_op)
+        # Only pass on the field names that are set in the current setup
+        instructions = set([name[:-4:] for name, _ in setup_op.iter_params()])
+        current_fields = {
+            key: val for key, val in acc_op.field_items() if key[:-4:] in instructions
+        }.items()
         # Create the sequence of all operations that need to be emitted
-        return combine_pairs_to_ops(acc_op.field_items(), vals, xcustom_acc)
+        return combine_pairs_to_ops(current_fields, vals, xcustom_acc)
 
 
-def combine_pairs_to_ops(field_items, values, xcustom_acc):
+def create_pairs(
+    fields_op: acc.LaunchOp | acc.SetupOp,
+) -> dict[str, tuple[SSAValue, SSAValue]]:
+    """
+    For a given RoCC launch or setup op, return a map that maps a single RoCC
+    instruction to pairs of two SSAValues
+
+    For setup ops, this can retrace back previous setup state if necessary
+    (i.e. if one of the two operands of an instruction gets dedupped)
+    """
+    # Make a set of all the unique instruction names in the current operation
+    instructions = set([name[:-4:] for name, _ in fields_op.iter_params()])
+    field_dict = dict(fields_op.iter_params())
+
+    # For setup_ops, get the previous setup state, if necessary
+    if isinstance(fields_op, acc.SetupOp):
+        prev_state = infer_state_of(fields_op.in_state) if fields_op.in_state else {}
+        for instruction in instructions:
+            if instruction + ".rs1" not in field_dict:
+                field_dict[instruction + ".rs1"] = prev_state[instruction + ".rs1"]
+            if instruction + ".rs2" not in field_dict:
+                field_dict[instruction + ".rs2"] = prev_state[instruction + ".rs2"]
+    # For launch_ops, no tracing back is necessary, since dedup doesn't happen
+    elif isinstance(fields_op, acc.LaunchOp):
+        pass
+    # Assert that pairs of rs1 and rs2 exist for each item in the fields
+    for instruction in instructions:
+        assert instruction + ".rs1" in field_dict, f"No rs1 found for {instruction}"
+        assert instruction + ".rs2" in field_dict, f"No rs2 found for {instruction}"
+    # Create a dictionary that contains the two vals associated
+    # to each single RoCC instruction
+    map: dict[str, tuple[SSAValue, SSAValue]] = {}
```

Comment:
can we change the variable name as to not confuse it with https://www.geeksforgeeks.org/python-map-function/